vertx-web-graphql/src/main/java/io/vertx/ext/web/handler/graphql/impl/GraphQLHandlerImpl.java

```diff
@@ -264,6 +270,67 @@ private void handlePostQuery(RoutingContext rc, GraphQLQuery query, String opera
     executeOne(rc, query);
   }
 
+  private void handlePostMultipart(RoutingContext rc, String operationName, Map<String, Object> variables) {
+    GraphQLInput graphQLInput;
+    try {
+      graphQLInput = parseApolloAttributes(rc);
+    } catch (Exception e) {
+      rc.fail(400, e);
+      return;
+    }
+
+    if (graphQLInput instanceof GraphQLQuery) {
+      handlePostQuery(rc, (GraphQLQuery) graphQLInput, operationName, variables);
+    } else {
+      rc.fail(500);
+    }
+  }
+
+  private GraphQLInput parseApolloAttributes(RoutingContext rc) {
+    final MultiMap attrs = rc.request().formAttributes();
+    final Map<String, List<String>> map = Json.decodeValue(attrs.get("map"),
+        new TypeReference<HashMap<String, List<String>>>() {});
+    final GraphQLQuery input = Json.decodeValue(attrs.get("operations"), GraphQLQuery.class);
+
+    final Map<String, Object> variables = new HashMap<>();
+    variables.put("variables", input.getVariables());
+
+    for (final Map.Entry<String, List<String>> entry : map.entrySet()) {
+      for (final String fullPath : entry.getValue()) {
+        final String[] path = fullPath.split("\\.");
+        int pathSize = path.length;
+
+        int idx;
+        try {
+          idx = Integer.parseInt(path[pathSize - 1]);
+          --pathSize;
+        } catch (final Exception e) {
+          idx = -1;
```

Comment:
Don't use exceptions to find if something isn't a number, it's very expensive.

Reply:
Done